### PR TITLE
[Task #1891334] Support create function app in Eclipse

### PR DESCRIPTION
**Pull request by Flanker32**
[AB#1891334](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1891334)

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.appservice/src/com/microsoft/azure/toolkit/eclipse/appservice/AppServiceCreationComposite.java [hunk 1]
```diff
@@ -33,48 +32,41 @@
...
-        setShellStyle(SWT.SHELL_TRIM);
-    }
+    private AppServicePlanComposite appServicePlanPanel;
 
-    @Override
-    protected Control createDialogArea(Composite parent) {
-        Composite container = (Composite) super.createDialogArea(parent);
+    public AppServiceCreationComposite(Composite parent, int style, Supplier<T> supplier) {
+        super(parent, style);
+        this.supplier = supplier;
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.appservice/src/com/microsoft/azure/toolkit/eclipse/appservice/AppServiceCreationComposite.java]: what's supplier for? the name is too general

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.appservice/src/com/microsoft/azure/toolkit/eclipse/appservice/AppServiceCreationComposite.java [hunk 2]
```diff
@@ -96,29 +88,38 @@ protected Control createDialogArea(Composite parent) {
...
-                .resourceGroup(subsAndResourceGroupPanel.getResourceGroup().getName())
-                .appName(instanceDetailPanel.getAppName())
-                .region(instanceDetailPanel.getResourceRegion())
-                .runtime(instanceDetailPanel.getRuntime())
-                .servicePlanName(entity.getName())
-                .servicePlanResourceGroup(StringUtils.firstNonBlank(entity.getResourceGroup(),
-                        subsAndResourceGroupPanel.getResourceGroup().getName()))
-                .pricingTier(entity.getPricingTier());
+        final T result = supplier.get();
+        result.subscriptionId(subsAndResourceGroupPanel.getSubscription().getId());
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.appservice/src/com/microsoft/azure/toolkit/eclipse/appservice/AppServiceCreationComposite.java]: Have you considered the case subsAndResourceGroupPanel.getSubscription() returns null

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.appservice/src/com/microsoft/azure/toolkit/eclipse/functionapp/EclipseFunctionAppActionsContributor.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import com.microsoft.azure.toolkit.lib.common.task.AzureTaskManager;
+
+public class EclipseFunctionAppActionsContributor implements IActionsContributor {
+    public static final int INITIALIZE_ORDER = FunctionAppActionsContributor.INITIALIZE_ORDER + 1;
+
+    @Override
+    public void registerHandlers(AzureActionManager am) {
+        final BiPredicate<Object, Object> createCondition = (r, e) -> r instanceof AzureFunction;
+        final BiConsumer<Object, Object> createHandler = (c, e) -> AzureTaskManager.getInstance()
+                .runLater(() -> CreateFunctionAppHandler.createFunctionApp());
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.appservice/src/com/microsoft/azure/toolkit/eclipse/functionapp/EclipseFunctionAppActionsContributor.java]: CreateFunctionAppHandler.createFunctionApp is strange, 
new CreateFunctionAppHandler().createFunctionApp seems better.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.appservice/src/com/microsoft/azure/toolkit/eclipse/functionapp/creation/CreateFunctionAppHandler.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        CreateFunctionAppDialog createDialog = new CreateFunctionAppDialog(shell, getDefaultAppServiceConfig());
+        createDialog.setOkActionListener(config -> {
+            createDialog.close();
+            AzureTaskManager.getInstance().runInBackground(AzureString.format("Creating function app %s", config.appName()),
+                    () -> new CreateOrUpdateFunctionAppTask(config).execute());
+        });
+        createDialog.open();
+    }
+
+    private static FunctionAppConfig getDefaultAppServiceConfig() {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.appservice/src/com/microsoft/azure/toolkit/eclipse/functionapp/creation/CreateFunctionAppHandler.java:45]: this method shall be in class FunctionAppConfig

Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.appservice/src/com/microsoft/azure/toolkit/eclipse/functionapp/creation/CreateFunctionAppHandler.java:45]: `FunctionAppConfig` is shared within different tooling and they may have different default values, for instance, we will provide default name for IDE but this is required for maven toolkits.

**Review by andxu — APPROVED** [review]
LGTM

**Pull request merged**